components/root/root.jsx

```diff
@@ -265,8 +265,8 @@ export default class Root extends React.PureComponent {
         } else {
             landing = desktopAppDownloadLink;
         }
-
-        if (landing && !BrowserStore.hasSeenLandingPage() && !toResetPasswordScreen && !this.props.location.pathname.includes('/landing') && !this.props.location.href?.includes('.test.mattermost.com') && !UserAgent.isDesktopApp()) {
+        const isTestServer = window.location.hostname?.search(/^.*\.test\.mattermost\.com$/g) !== 0;
```

Comment:
Do we need the regex here? Couldn't we write this as `window.location.hostname.endsWith('.test.mattermost.com')`?

Unlike `this.props.location`, I don't think `window.location.hostname` can ever be undefined based on [this](https://developer.mozilla.org/en-US/docs/Web/API/Location/hostname), so we might not need the `?.` either

Reply:
@hmhealey yeah right it couldn’t be empty in the browser but in the unit test it will be empty in fact test was failing due to that.
About using regex I used regex as per this suggestion and To me, it did make sense https://github.com/mattermost/mattermost-webapp/pull/10900#discussion_r940932227

Reply:
I tested it out, and it looks like `const isTestServer = window.location.hostname?.endsWith('.test.mattermost.com');` is accepted by CodeQL. I think the reason it was complaining about the previous version was more that we were looking at the entire URL with `location.href` instead of just the hostname like we're doing now.

Reply:
```suggestion
        const isTestServer = window.location.hostname?.startsWith('.test.mattermost.com');
```